monkey/infection_monkey/master/plugin_registry.py

```diff
@@ -1,17 +1,39 @@
 import logging
+from copy import copy
+from pathlib import Path
+from tarfile import TarFile
+from threading import current_thread
 from typing import Any
 
 from serpentarium import PluginLoader
 
 from common.agent_plugins import AgentPluginType
 from infection_monkey.i_puppet import UnknownPluginError
-from infection_monkey.island_api_client import IIslandAPIClient
+from infection_monkey.island_api_client import IIslandAPIClient, IslandAPIRequestError
 
 logger = logging.getLogger()
 
 
+def check_safe_archive(dest_path: Path, archive: TarFile) -> bool:
+    path = Path.resolve(dest_path)
+    for name in archive.getnames():
+        if path not in Path.resolve(dest_path / name).parents:
+            return False
```

Comment:
We also need to assert that the file `isfile()` or  `isdir()`. We don't want to extract links, or devices. Don't forget to add tests.